tests/result/test_stream_results.py

```diff
@@ -0,0 +1,87 @@
+""" Test StreamArray """
+from unittest.mock import patch
+
+import numpy as np
+import pytest
+
+from qililab.result import stream_results
+from qililab.result.stream_results import StreamArray
+
+AMP_VALUES = np.arange(0, 1, 2)
+
+
+@pytest.fixture(name="stream_array")
+def fixture_stream_array():
+    """fixture_stream_array
+
+    Returns:
+        stream_array: StreamArray
+    """
+    shape = (2, 2)
+    path = "test_stream_array.hdf5"
+    loops = {"test_amp_loop": AMP_VALUES}
+
+    return stream_results(shape=shape, path=path, loops=loops)
+
+
+class MockGroup:
+    """Mock a h5py group."""
+
+    def create_dataset(self, name: str, data: np.ndarray):  # pylint: disable=unused-argument
+        """Creates a dataset"""
+        return {}
+
+
+class MockFile:
+    """Mocks a h5py file."""
+
+    def __init__(self):
+        """Initialize a mock file."""
+        self.dataset = None
+
+    def create_group(self, name: str):  # pylint: disable=unused-argument
+        return MockGroup()
+
+    def create_dataset(self, name: str, data: np.ndarray):  # pylint: disable=unused-argument
+        """Creates a dataset"""
+        return data
```

Comment:
You should return a `copy.deepcopy` of this!

Reply:
This is because h5py creates a copy of the data before putting it inside the dataset